### DISCUSSION TITLE
Server is slow: what about using WebRT?

### DISCUSSION THREAD

**Discussion by francoisschwarzentruber** [tableaunoir/tableaunoir#115] [Ideas]
Issue:
The current server tableaunoir.irisa.fr has 1Go of RAM and a small CPU (I have to ask to the IT team). There are 2000 persons per week, that is not lot but I expect that some connections are made at the same time so the server slows down, or shuts down.

Context:
Anyway, the philosophy of tableaunoir is to have ephemera boards, boards are not saved, and you have to save them yourself.

Solution:
To use WebRT (Peer-to-peer technology)! The server could be used only for making the connection between people, verify some privileges etc. Then if the server is slow, that is ok.

What do you think?

**Comment by tekhnee** [comment]
@francoisschwarzentruber Your server specs are indeed minimal. Kudos for making tableaunoir work so well on such resources!

Not having expertise in WebRTC, I've seen it work well with small groups (e.g. ZipCall). But I wonder whether this peer-to-peer topology would scale as well with larger numbers of interacting participants (10 or more). My question is arbitrary because, again, I have never grappled with these particular issues. Thanks.

**Comment by tekhnee** [comment]
I've long been following discussions about [NextCloud Talk](https://nextcloud.com/talk/), because we deploy a NextCloud instance here and Talk would make a natural addition to it. At the no-fee pricing tier, talk is a WebRTC solution and, unfortunately, the forums are rife with complaints about its problematic scaling by comparison with, say, Jitsi, which automatically switches from WebRTC P2P to a "videobridge" topology when necessary. (According to reports, Talk P2P begins to fall apart with >5 participants, more or less.) That said, these are all videoconferencing platforms, with bandwidth requirements far higher than those of tableaunoir…
